aspnetcore/performance/rate-limit.md

```diff
@@ -19,6 +19,154 @@ The `Microsoft.AspNetCore.RateLimiting` middleware provides rate limiting middle
+  
+  builder.Services.AddRateLimiter(options =>
+  {
+      options.AddFixedWindowLimiter("fixed", opt =>
+      {
+          opt.PermitLimit = 4;
+          opt.Window = TimeSpan.FromSeconds(12);
+          opt.QueueProcessingOrder = QueueProcessingOrder.OldestFirst;
+          opt.QueueLimit = 2;
+      });
+  });
+  
+  var app = builder.Build();
+  ```
+  
+  The global limiter applies to all endpoints automatically when it's configured via [options.  GlobalLimiter](/dotnet/api/microsoft.aspnetcore.ratelimiting.ratelimiteroptions.globallimiter), and no endpoint-specific policy is specified.
+
+3. Enable rate limiting middleware
+
+   In the `Program.cs` file, enable the rate limiting middleware by calling [UseRateLimiter](/dotnet/api/microsoft.aspnetcore.builder.ratelimiterapplicationbuilderextensions.useratelimiter):
+  
+  ``` csharp
+  app.UseRouting();
+  
+  app.UseRateLimiter();
+  
+  app.UseEndpoints(endpoints =>
+  {
+      endpoints.MapControllers();
+  });
+  
+  app.Run();
+  ```
+
+### Apply rate limiting policies to endpoints or pages
+
+#### Apply rate limiting to WebAPI Endpoints
+
+Apply a named policy to the endpoint or group, for example:
+
+``` csharp
+
+app.MapGet("/api/resource", () => "This endpoint is rate limited")
+   .RequireRateLimiting("fixed"); // Apply specific policy to an endpoint
+
+```
+
+#### Apply rate limiting to MVC Controllers
+
+ Apply the configured rate limiting policies to specific endpoints or globally. For example, to apply the "fixed" policy to all controller endpoints:
+
+``` csharp
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapControllers().RequireRateLimiting("fixed");
+});
+
+```
+
+#### Apply rate limiting to server-side Blazor apps
+
+To set rate limiting to all pages,  [`RequireRateLimiting(Policy)`](/dotnet/api/microsoft.aspnetcore.builder.ratelimiterendpointconventionbuilderextensions.requireratelimiting) can be specified on the MapRazorComponents call, for example:
+
+``` csharp
+app.MapRazorComponents<App>()
+    .AddInteractiveServerRenderMode()
+    .RequireRateLimiting("Policy");
+```
+
+To set a policy for a single Razor component, the [`[EnableRateLimiting]` attribute](xref:Microsoft.AspNetCore.RateLimiting.EnableRateLimitingAttribute) is applied to the component with an optional policy name:
+
+``` blazor
+@page "/counter"
+@using Microsoft.AspNetCore.RateLimiting
+@attribute [EnableRateLimiting("Policy")]
+
+<h1>Counter</h1>
+```
+
+The [`DisableRateLimiting`](/dotnet/api/microsoft.aspnetcore.ratelimiting.disableratelimitingattribute) attribute can be used to disable rate limiting on a Razor Page.
+
+The [`[EnableRateLimiting]` attribute](xref:Microsoft.AspNetCore.RateLimiting.EnableRateLimitingAttribute) is only applied to a component if <xref:Microsoft.AspNetCore.Builder.RateLimiterEndpointConventionBuilderExtensions.RequireRateLimiting%2A> is ***not*** called on <xref:Microsoft.AspNetCore.Builder.RazorComponentsEndpointRouteBuilderExtensions.MapRazorComponents%2A>.
```

Comment:
The order of Line 166 and 168 should be swapped because Line 168 is a continuation of the preceding information on the `[EnableRateLimiting]` attribute.